clinica/iotools/converters/aibl_to_bids/utils/clinical.py

```diff
@@ -168,263 +208,119 @@ def create_sessions_tsv_file(
-
-                elif field in list(df.columns.values) and field == "EXAMDATE":
-                    exam_date = df.loc[(df["RID"] == rid), field]
-
-                elif field in list(df.columns.values) and field == "PTDOB":
-                    patient_date_of_birth = df.loc[(df["RID"] == rid), field]
-
-        exam_dates = _clean_exam_dates(
-            rid, exam_date.to_list(), visit_code.to_list(), clinical_data_dir
-        )
+    for bids_id in get_bids_subjs_list(bids_dir):
+        rid = int(bids_id_factory(study)(bids_id).to_original_study_id())
+        sessions = pd.DataFrame(
+            {"session_id": get_bids_sess_list(bids_dir / bids_id)}
+        ).set_index("session_id", drop=False)
 
-        if not patient_date_of_birth.empty:
-            age = _compute_ages_at_each_exam(
-                patient_date_of_birth.values[0], exam_dates
+        for _, row in specifications.iterrows():
+            df = pd.read_csv(
+                glob.glob(str(clinical_data_dir / row[f"{study} location"]))[0],
+                dtype={"text": str},
             )
-        else:
-            age = "n/a"
-
-        visit_code[visit_code == "bl"] = "M000"
-        visit_code = visit_code.str.upper()
+            extract = _extract_metadata_df(
+                input_df=df,
+                source_id=rid,
+                bids_metadata=row["BIDS CLINICA"],
+                source_metadata=row[study],
+            )
+            sessions = pd.concat([sessions, extract], axis=1)
 
-        sessions = pd.DataFrame(
-            {
-                "months": visit_code.str[1:],
-                "age": age,
-                "MMS": mm_score,
-                "cdr_global": cd_global,
-                "diagnosis": dx_curren,
-                "examination_date": exam_dates,
-            }
+        sessions.sort_index(inplace=True)
+        # -4 are considered missing values in AIBL
+        sessions.replace([-4, "-4", np.nan], None, inplace=True)
+        sessions["diagnosis"] = sessions.diagnosis.apply(
+            lambda x: _mapping_diagnosis(x)
         )
-        # todo (LATER) : pretty sure there exists a function that converts viscode to session
-        sessions = sessions.assign(
-            session_id=lambda df: df.months.apply(lambda x: f"ses-M{int(x):03d}")
+        sessions["examination_date"] = sessions.apply(
+            lambda x: _complete_examination_dates(
+                rid, x.session_id, x.examination_date, clinical_data_dir
+            ),
+            axis=1,
         )
-        cols = sessions.columns.tolist()
-        sessions = sessions[cols[-1:] + cols[:-1]]
 
-        bids_id = bids_id_factory(StudyName.AIBL).from_original_study_id(str(rid))
+        # in general age metadata is present only for baseline session
+        sessions["age"] = sessions["age"].ffill()
+        sessions["age"] = sessions.apply(
+            lambda x: _compute_age_at_exam(x.age, x.examination_date), axis=1
```

Comment:
I'm probably missing something but aren't you supposed to pass the date of birth as the first parameter ?

```python
def _compute_age_at_exam(birth_date: Optional[str], exam_date: Optional[str])
```

Reply:
Yes totally !